plugins/catalog-backend-module-github/src/lib/github.ts

```diff
@@ -18,11 +18,24 @@ import { GroupEntity, UserEntity } from '@backstage/catalog-model';
 import { GithubCredentialType } from '@backstage/integration';
 import { graphql } from '@octokit/graphql';
 
+/**
+ * Generic mapper for mapping input type to output type
+ *
+ * @public
+ */
+export interface IMapper<InputType, OutputType> {
```

Comment:
We avoid prefixing types with `I`. Also see comment below about using the terminology "transformer" instead of "mapper".

This is what the corresponding definitions look like in the ldap module: https://github.com/backstage/backstage/blob/6389362f0dcbdf0e7cea34b0ada6268bcf98e0de/plugins/catalog-backend-module-ldap/src/ldap/types.ts#L21

They keep it simple by requiring that it returns a promise (it's pretty ok, you can just slap an `async` on the custom function and you're done), but I don't have a strong opinion on that one honestly.

They also pass in some extra data, such as the config of the integration that was used at that point. This can be useful since sometimes you have multiple sources and want to do different things (such as adding different annotations) depending on what source you're reading currently.

So maybe name this one `UserTransformer` as well, so there's good room for a (possibly slightly different) `GroupTransformer` next to it